apis/duck/typed_test.go

```diff
@@ -94,6 +95,26 @@ func TestSimpleList(t *testing.T) {
 	// TODO(mattmoor): Access through informer
 }
 
+func TestInvalidResource(t *testing.T) {
+	client := &invalidResourceClient{}
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	tif := &duck.TypedInformerFactory{
+		Client:       client,
+		Type:         &duckv1alpha1.AddressableType{},
+		ResyncPeriod: 1 * time.Second,
+		StopChannel:  stopCh,
+	}
+
+	want := errors.New("failed to get list")
+	_, _, got := tif.Get(SchemeGroupVersion.WithResource("resources"))
+
+	if got == nil || got.Error() != want.Error() {
```

Comment:
you can do this easier, by using a variable (global to the test, or passing it to the invalid resource, up to you), so that it is returned from the call. 
That way you can just do
```
var testErr = errors.New("...")
...
if got != testErr {
  t.Errorf(....)
}
```